### PR TITLE
Add depth input

### PR DESCRIPTION
**Pull request by melisandeteng**
We add depth input: 
- depth map are read as images
- set as input as normalized depth maps [-1, 1]
Depth maps can come in the encoding of our simulator or as images from Megadepth model.

Not handled yet: displaying the depth maps in the log

### CODE REVIEW

File: data/datasets.py [hunk 1]
```diff
@@ -228,26 +224,222 @@ def __len__(self):
...
-            RealSimDataset(
-                opts,
-                transform=transforms.Compose(get_transforms(Dict(opts))),
-                no_check=no_check,
-            ),
-            batch_size=opts.data.loaders.get("batch_size", 4),
-            shuffle=opts.model.is_train,
-            num_workers=opts.data.loaders.get("num_workers", 8),
-        )
+        if depth is True:
```
Comment (vict0rsch) [inline data/datasets.py:229]: why not just `if depth`?

File: data/datasets.py [hunk 2]
```diff
@@ -228,26 +224,222 @@ def __len__(self):
...
-        return DataLoader(
-            SimDataset(
-                opts,
-                transform=transforms.Compose(get_transforms(Dict(opts))),
-                no_check=no_check,
-            ),
-            batch_size=opts.data.loaders.get("batch_size", 4),
-            shuffle=opts.model.is_train,
-            num_workers=opts.data.loaders.get("num_workers", 8),
+        if depth is True:
```
Comment (vict0rsch) [inline data/datasets.py:252]: idem

File: data/datasets.py [hunk 3]
```diff
@@ -228,26 +224,222 @@ def __len__(self):
...
+            return json.load(f)
+
+    def yaml_load(self, file_path):
+        with open(file_path, "r") as f:
+            return yaml.safe_load(f)
+
+    def __getitem__(self, i):
+        """Return an item in the dataset with fields:
+        {
+            data: transform({),
```
Comment (vict0rsch) [inline data/datasets.py:326]: typo?

File: data/datasets.py [hunk 4]
```diff
@@ -228,26 +224,222 @@ def __len__(self):
...
+            np.array(datadict["rd"]), mode=self.opts.depth.real_mode
+        )
+
+        return Dict({"data": datadict, "paths": paths})
+
+    def __len__(self):
+        return len(self.samples_paths)
+
+
+class SimDepthDataset(Dataset):
```
Comment (vict0rsch) [inline data/datasets.py:372]: What is so different between the `RealSimDepth` and `SimDepth` datasets that we need to have almost duplicate code? Can't we just use a few `if`s? 

I know it's the cas for non-depth datasets, maybe that should change too. It's very error-prone IMHO

**Comment by vict0rsch** [comment]
I don't understand why there are so many generator files. It sounds like this is going to be very costly to maintain as features are added to the architectures...

File: data/datasets.py (hunk 4 above)
Comment (melisandeteng) [reply data/datasets.py:372]: Yes I was tempted not to put any SimDepth dataset as I noticed even the rest of the code for non depth datasets is actually  not meant to be used without real data also the dataset without real exists,... 
I'll see what I can do about this

File: data/datasets.py (hunk 3 above)
Comment (melisandeteng) [reply data/datasets.py]: yes, probably from the code of the non depth classes

**Comment by melisandeteng** [comment]
Accidentally pushed sim_mask_depth_generator.py and  sim_mask_generator.py indeed. There should just be mask_generator and mask_depth_generator  @vict0rsch
